test/test_mongo_middleware.js

```diff
@@ -238,6 +240,48 @@ describe('mongo db middleware', function() {
       });
     });
 
+    it('passes isForSubmit = true in options when fields has $submit = true', function(done) {
+      sinon.spy(Collection.prototype, 'find');
```

Comment:
Replacing a prototype method with a spy in this manner can cause test weirdness if this test case fails, since the original won't be restored if the test never gets to line 279.

To help with that, Sinon has a handy sandbox feature, where you create a "sandbox" in `beforeEach`, use the sandbox to create the spy, and then restore the sandbox in `afterEach`, thus ensuring the restore happens even if the test case fails:
https://sinonjs.org/releases/latest/spies/#using-a-spy-to-wrap-all-object-method

You can do the sandbox management in the top-level describe block's `beforeEach` and `afterEach`, that way all test cases in the file will have the sandbox to use if they want to.

Reply:
(in our organisation's tests, we have a global `afterEach()` hook that calls `sinon.restore()` to make sure we always reset our stubs everywhere)